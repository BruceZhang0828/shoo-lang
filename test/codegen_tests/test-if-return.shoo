function foo() string {
    string x = "Hello "+"World"+"!";
    string y = "what";
    if (x!=y) {
        return "correct";
    } else {
        return "incorrect";
    }
}
println(foo());