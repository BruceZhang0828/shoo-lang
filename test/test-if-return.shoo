function foo() string {
    string x = "Hello "+"World"+"!";
    string y = "what";
    if (x!=y) {
        return "correct";
    } else {
        return "incorrect";
    }
}
println(foo());

function bar() string {
    float x = 1.0909;
    float y = 2.898;
    if (x==y) {
        return "incorrect";
    } elif (x > y) {
        return "incorrect";
    } elif (x < y) {
        return "correct";
    } else {
        return "incorrect";
    }
}
println(bar());


function someReturn(float a) string {
    if (a == 25.9) {
        return "okay";
    } else {
        
    }
    
    return "maybe";
}

someReturn(25.9);

function other(bool b) int {
    int a = 1;
    if (b) {
        a++;
    } elif (false) {
        a = 0;
    } else {
        return a + other(true);
    }
    return a;
}

println(str_of_int(other(true)));
println(str_of_int(other(false)));
