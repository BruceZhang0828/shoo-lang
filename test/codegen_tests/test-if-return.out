correct
correct
2
3
